src/ansys/sound/core/sound_composer/source_spectrum.py

```diff
@@ -0,0 +1,239 @@
+
+        # load_source_spectrum can only be called after __operator_load is defined.
+        if len(file_source) > 0:
+            self.load_source_spectrum(file_source)
+        else:
+            self.source_spectrum_data = None
+
+    def __str__(self) -> str:
+        """Return the string representation of the object."""
+        if self.source_spectrum_data is not None:
+            support_data = self.source_spectrum_data.time_freq_support.time_frequencies.data
+            str_source = (
+                f"'{self.source_spectrum_data.name}'\n"
+                f"\tFmax: {support_data[-1]} Hz\n"
+                f"\tDeltaF: {support_data[1] - support_data[0]} Hz"
+            )
+        else:
+            str_source = "Not set"
+
+        if self.is_source_control_set():
+            str_source_control = (
+                f"{self.source_control.get_method_name()}, {self.source_control.duration} s"
+            )
+        else:
+            str_source_control = "Not set"
+
+        return f"Spectrum source: {str_source}\nSource control: {str_source_control}"
+
+    @property
+    def source_control(self) -> SourceControlSpectrum:
+        """Spectrum source control object.
+
+        Contains the duration and the method used to generate the sound.
+        """
+        return self.__source_control
+
+    @source_control.setter
+    def source_control(self, source_control: SourceControlSpectrum):
+        """Set the spectrum source control."""
+        if not (isinstance(source_control, SourceControlSpectrum) or source_control is None):
+            raise PyAnsysSoundException(
+                "Specified source control object must be of type ``SourceControlSpectrum``."
+            )
+        self.__source_control = source_control
+
+    @property
+    def source_spectrum_data(self) -> Field:
+        """Spectrum source data as a DPF field.
+
+        Contains the frequencies, the levels (as a PSD) and the unit (unit^2/Hz).
+        """
+        return self.__source_spectrum_data
+
+    @source_spectrum_data.setter
+    def source_spectrum_data(self, source_spectrum_data: Field):
+        """Set the spectrum source as a DPF field."""
+        if not (isinstance(source_spectrum_data, Field) or source_spectrum_data is None):
+            raise PyAnsysSoundException(
+                "Specified spectrum source must be provided as a DPF field."
+            )
+        self.__source_spectrum_data = source_spectrum_data
+
+    def is_source_control_set(self) -> bool:
+        """Source control verification.
+
+        Check if the spectrum source control is set and its duration is strictly positive.
+
+        Returns
+        -------
+        bool
+            True if the spectrum control is set and its duration is strictly positive, False
+            otherwise.
+        """
+        return (self.source_control is not None) and (self.source_control.duration > 0.0)
```

Comment:
@ansaminard the second check is probably redundant since the setter for `duration` in `SourceControlSetter` already ensures that it's > 0.0

No need to remove it but it's probably not needed :)

Reply:
@a-bouth Yes, we've discussed this with @anshlachamb, but there is a slight difference here, as we want the duration to be _strictly_ greater than 0. In the duration setter on the other hand, we only do a "loose" comparison because the duration is initiated at 0 s in the ctor (and according @anshlachamb we want to keep it this way).
It's because of this difference that we decided to keep the duration check here too, despite the apparent redundancy.